From c8e67008d6c9cb6a3bdfbe76915caa8306d1466a Mon Sep 17 00:00:00 2001
From: Colin Xu <colin.xu@intel.com>
Date: Fri, 17 Apr 2020 11:04:08 +0800
Subject: [PATCH 10/24] drm/i915/gvt: Add more mmios to non-priv whilte list
 for CFL-R

Guest driver will write these mmios to force non-priv regs.

Signed-off-by: Colin Xu <colin.xu@intel.com>
(cherry picked from commit 9fa10155457db42b9df604399ae6f92eff7857d0)
Signed-off-by: Colin Xu <colin.xu@intel.com>
(cherry picked from commit b2d9bd22c28ba300a3f657b0401f6445989e1c0f)
Signed-off-by: Colin Xu <colin.xu@intel.com>
---
 drivers/gpu/drm/i915/gvt/handlers.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/gvt/handlers.c b/drivers/gpu/drm/i915/gvt/handlers.c
index 8bec33ba5976..41e57939307e 100644
--- a/drivers/gpu/drm/i915/gvt/handlers.c
+++ b/drivers/gpu/drm/i915/gvt/handlers.c
@@ -532,6 +532,8 @@ static int pipeconf_mmio_write(struct intel_vgpu *vgpu, unsigned int offset,
 static i915_reg_t force_nonpriv_white_list[] = {
 	GEN9_CS_DEBUG_MODE1, //_MMIO(0x20ec)
 	GEN9_CTX_PREEMPT_REG,//_MMIO(0x2248)
+	_MMIO(0x2341),
+	_MMIO(0x2351),
 	GEN8_CS_CHICKEN1,//_MMIO(0x2580)
 	_MMIO(0x2690),
 	_MMIO(0x2694),
@@ -556,6 +558,10 @@ static i915_reg_t force_nonpriv_white_list[] = {
 	_MMIO(0xe18c),
 	_MMIO(0xe48c),
 	_MMIO(0xe5f4),
+	_MMIO(0x10000d82),
+	_MMIO(0x10002349),
+	_MMIO(0x10064844),
+	_MMIO(0x4000b118),
 };
 
 /* a simple bsearch */
-- 
2.17.1

